packages/network-controller/src/NetworkController.ts

```diff
@@ -63,32 +73,178 @@ export type NetworkMetadata = {
+  /**
+   * The URL of the endpoint. Expected to be a sort of template with the text
+   * `{infuraProjectId}`, which will get replaced with the Infura project ID.
+   */
+  url: string;
+};
+
+/**
+ * A custom RPC endpoint is a reference to a user-defined server which fronts an
+ * EVM chain. It may refer to an Infura network, but only by coincidence.
+ */
+export type CustomRpcEndpoint = {
+  /**
+   * The user-facing name of the endpoint.
+   */
+  name: string;
+  /**
+   * The identifier for the network client that has been created for this RPC
+   * endpoint.
+   */
+  networkClientId: string;
+  /**
+   * The type of this endpoint, always "custom".
+   */
+  type: RpcEndpointType.Custom;
+  /**
+   * The URL of the endpoint.
+   */
+  url: string;
+};
+
+/**
+ * An RPC endpoint is a reference to a server which fronts an EVM chain. There
+ * are two varieties of RPC endpoints: Infura and custom.
+ *
+ * @see CustomRpcEndpoint
+ * @see InfuraRpcEndpoint
+ */
+export type RpcEndpoint = InfuraRpcEndpoint | CustomRpcEndpoint;
+
+/**
+ * From a user perspective, a network configuration holds information about a
+ * network that a user can select through the client. A "network" in this sense
+ * can explicitly refer to an EVM chain that the user explicitly adds or doesn't
+ * need to add (because it comes shipped with the client). The properties here
+ * therefore directly map to fields that a user sees and can edit for a network
+ * within the client.
  *
- * @property rpcUrl - RPC target URL.
- * @property chainId - Network ID as per EIP-155
- * @property nickname - Personalized network name.
- * @property ticker - Currency ticker.
- * @property rpcPrefs - Personalized preferences.
+ * Internally, a network configuration represents a single conceptual EVM chain,
+ * which is represented tangibly via multiple RPC endpoints. A "network" is then
+ * something for which a network client object is created automatically or
+ * created on demand when it is added to the client.
  */
 export type NetworkConfiguration = {
-  rpcUrl: string;
+  /**
+   * An optional URL that allows the user to view blocks and transactions that
+   * have occurred on the chain.
+   */
+  blockExplorerUrl?: string;
+  /**
+   * The ID of the chain. Represented in hexadecimal format with a leading "0x"
+   * instead of decimal format so that when viewed out of context it can be
+   * unambiguously interpreted.
+   */
   chainId: Hex;
-  ticker: string;
-  nickname?: string;
-  rpcPrefs?: {
-    blockExplorerUrl: string;
-  };
+  /**
+   * The RPC endpoint URL that all requests will use by default in order to
+   * interact with the chain. The value of this property must match the `url`
+   * property of an entry in `rpcEndpoints`.
+   */
+  defaultRpcEndpointUrl: string;
+  /**
+   * The user-facing nickname assigned to the chain.
+   */
+  name: string;
+  /**
+   * The name of the token that represents the native currency for the chain.
+   */
+  nativeTokenName: string;
```

Comment:
This property used to be called `ticker`. I decided to rename it because although "ticker" makes sense from the perspective of trading, I feel that it's more of a colloquial term. I believe it's more technically accurate to say that each chain has a native token, and that token has a name. So that's why I decided to rename this. That said, if we feel that `ticker` isn't confusing, I'm okay with sticking with it. Thoughts?

Reply:
see https://github.com/MetaMask/core/pull/4286#discussion_r1630102068

Reply:
It's not correct to call this a "token" though. It's a currency. The term we use for it elsewhere is "native currency".

We use the term "tokens" typically for on-chain assets like ERC-20, ERC-721, and ERC-1155. These all provide a "token" that is represented on-chain. But no such thing exists for currency. Similarly, we wouldn't describe real-life currency like USD as "tokens".

Reply:
Change made here: [98d4c2a](https://github.com/MetaMask/core/pull/4286/commits/98d4c2a0269e9897453616c51610a66c3f308472)

Reply:
Resolving this conversation, but feel free to reopen if you feel differently.